experimental/rsocket-test/RSocketClientServerTest.cpp

```diff
@@ -1,32 +1,61 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
+#include <random>
 #include <utility>
 
-#include <folly/io/IOBuf.h>
 #include <gmock/gmock.h>
 
 #include "handlers/HelloStreamRequestHandler.h"
 #include "rsocket/RSocket.h"
 #include "rsocket/transports/TcpConnectionAcceptor.h"
+#include "rsocket/transports/TcpConnectionFactory.h"
 
 using namespace rsocket;
 using namespace rsocket::tests;
 
+namespace {
+
+std::random_device device;
+std::uniform_int_distribution<uint16_t> dis(9000, 10000);
+
+// Helps prevent against port collisions.
+uint16_t randPort() {
```

Comment:
Neat, first time I've heard of it.  I tried to change the start-shutdown test to do that (since I wasn't sending anything across the wire anyway), but it caused the test to crash.  I'm guessing we're depending on some `if (socket)` check somewhere...

I'll merge this as is, and look into the crashes separately.